src/casync-http.c

```diff
@@ -521,10 +526,17 @@ static int run(int argc, char *argv[]) {
                 if (arg_verbose)
                         log_info("Acquiring %s...", url_buffer);
 
-                if (curl_easy_perform(curl) != CURLE_OK) {
```

Comment:
If the connect error is not transient this implementation will retry forever? Maybe it is better to create a wrapper function around curl_easy_perform() that checks for CURLE_COULDNT_CONNECT and possibly other errors and retries the operation for a fixed number of times.

Reply:
> If the connect error is not transient this implementation will retry forever?

Yes :)

Indeed better to have a counter to retry a limited number of time.

I also noticed afterwards that `curl_easy_perform()` is called in other places of the code, so a wrapper like you suggest makes sense.

Reply:
Another idea could be to have an incremental delay between each retry. Here I used a dirty `sleep(1)` because, in my tests, when I get the `COULDNT_CONNECT` error, it takes a while before the server recovers.

Reply:
@mihais I updated the PR, I now use a wrapper around `curl_easy_perform` as you suggested. Much better indeed.